packages/app/components/stack-modal/StackModal.tsx

```diff
@@ -32,100 +31,226 @@ import {
+            <StackInfo stackOrder={stackOrder} />
+            {totalStackOrdersDone(stackOrder) > 0 && (
+              <StackOrdersTable stackOrder={stackOrder} />
+            )}
+          </div>
+        </ModalContent>
+        <ModalFooter>
+          {!stackOrder.cancelledAt && !stackIsComplete && (
+            <Button
+              size="sm"
+              action="secondary"
+              onClick={() => setIsOpenCancellationDialog(true)}
+              width="full"
+            >
+              Cancel Stacking
+            </Button>
           )}
-        </div>
-      </ModalContent>
-      <ModalFooter>
-        {!stackIsComplete && (
-          <Button
-            size="sm"
-            action="secondary"
-            onClick={() => alert("Are you sure you want to cancel stacking?")}
-            width="full"
-          >
-            Cancel Stacking
-          </Button>
+        </ModalFooter>
+      </Modal>
+      <Dialog
+        isOpen={isOpenCancellationDialog}
+        closeAction={() => setIsOpenCancellationDialog(false)}
+      >
+        <DialogContent
+          title=" Are you sure you want to cancel stacking?"
+          description={`The ${remainingFundsDescription}, will be sent to your wallet`}
+        />
+        <DialogFooterActions
+          primaryAction={() => cancelStack()}
+          primaryText="Proceed"
+          secondaryAction={() => setIsOpenCancellationDialog(false)}
+          secondaryText="Cancel"
+        />
+      </Dialog>
+      <DialogConfirmTransactionLoading
+        isOpen={isCancellationProcessing}
+        title={cancellationTx && "Proceeding cancellation"}
+        description={cancellationTx && "Waiting for transaction confirmation."}
+      >
+        {cancellationTx && cancellationTx.hash && (
+          <CancelTransactionLink txHash={cancellationTx.hash} />
+        )}
+      </DialogConfirmTransactionLoading>
+      <Dialog
+        isOpen={isCancellationSuccess}
+        closeAction={() => setIsCancellationSuccess(false)}
+      >
+        <Icon name="check" className="text-primary-400" size={38} />
+        <DialogContent
+          title="Stack Cancelled"
+          description={`The ${remainingFundsDescription}, were sent to your wallet`}
+        />
+        {cancellationTx && cancellationTx.hash && (
+          <CancelTransactionLink txHash={cancellationTx.hash} />
         )}
-      </ModalFooter>
-    </Modal>
+        <DialogFooterActions
+          primaryAction={() => {
+            refetchStacks();
+            setIsCancellationSuccess(false);
+            setIsCancellationProcessing(false);
+            closeAction();
+          }}
+          primaryText="Back to Stacks"
+        />
+      </Dialog>
+    </div>
   );
 };
 
+const CancelTransactionLink = ({ txHash }: { txHash: string }) => (
+  <a
+    className="capitalize text-primary-100"
+    href={txEplorerLink(txHash)}
+    target="blank"
+  >
+    check transaction
```

Comment:
Adding a style to something static and that we can change feels weird. We adding "more code" where we can change it manually 😄

Reply:
Also, should we had an external link icon here?